src/test/java/th/ac/kmitl/science/comsci/example/nameprinters/NamePrinterLoaderTest.java

```diff
@@ -0,0 +1,61 @@
+package th.ac.kmitl.science.comsci.example.nameprinters;
+
+import org.junit.Test;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+
+public class NamePrinterLoaderTest {
+
+    private Set<Class<? extends NamePrinter>> createSetOfNamePrinters() {
+        Set<Class<? extends NamePrinter>> mockNamePrinterClasses = new HashSet<>();
+
+        mockNamePrinterClasses.add(NamePrinter.class);
+        mockNamePrinterClasses.add(NamePrinterBase.class);
+        mockNamePrinterClasses.add(ChayapolNamePrinter.class);
+
+        return mockNamePrinterClasses;
+    }
+
+    private NamePrinterLoader setupMockNamePrinterLoader() {
+        NamePrinterLoader namePrinterLoader = NamePrinterLoader.getInstance();
+        namePrinterLoader.setAllNamePrinterClasses(createSetOfNamePrinters());
+        return namePrinterLoader;
+    }
+
+    @Test
+    public void setAllNamePrinterClasses(){
+        Set<Class<? extends NamePrinter>> namePrinterClasses = createSetOfNamePrinters();
+        NamePrinterLoader.getInstance().setAllNamePrinterClasses(namePrinterClasses);
+        assertEquals(namePrinterClasses, NamePrinterLoader.getInstance().getAllNamePrinterClasses());
+    }
+
+    @Test
+    public void getNamePrinters() {
+        int namePrinterCount = 0;
+        NamePrinterLoader namePrinterLoader = setupMockNamePrinterLoader();
+
+        for(NamePrinter namePrinter : namePrinterLoader)
```

Comment:
This loop can be archived by `namePrinter.size()` List

referred to: https://docs.oracle.com/javase/7/docs/api/java/util/List.html#size()

Reply:
@khemtat `namePrinter` is `NamePrinter` from `namePrinterLoader` which is `Iterable<NamePrinter>` that has no `size()` method. I'll add `count()` method to `NamePrinterLoader` to get `namePrinters.size()` because I don't need to expose `namePrinters`.

Reply:
ok krub 👍